src/panel/favorites/FavoritesPanel.jsx

```diff
@@ -0,0 +1,96 @@
+/* globals _ */
+import React from 'react';
+import Telemetry from 'src/libs/telemetry';
+import Panel from 'src/components/ui/Panel';
+import FavoriteItems from './FavoriteItems';
+import FavoriteMasqFooter from './FavoriteMasqFooter';
+import { version } from 'config/constants.yml';
+import MasqOnboardingModal from 'src/modals/masq_onboarding_modal';
+import Store from 'src/adapters/store';
+import PoiStore from 'src/adapters/poi/poi_store';
+import nconf from '@qwant/nconf-getter';
+import classNames from 'classnames';
+
+const isMasqEnabled = nconf.get().masq.enabled;
+
+export default class FavoritesPanel extends React.Component {
+  state = {
+    isLoggedIn: false,
+    favoritePois: [],
+    isReady: false,
+    masqFooterClosed: localStorage.getItem(`qmaps_v${version}_favorite_masq_footer`) !== 'false',
+  };
+
+  componentDidMount() {
+    Telemetry.add(Telemetry.FAVORITE_OPEN);
+    this.store = new Store();
+    this.store.onToggleStore(this.loadData);
+    this.loadData();
+  }
+
+  loadData = async () => {
+    const isLoggedIn = await this.store.isLoggedIn();
+    const favoritePois = await PoiStore.getAll();
+    this.setState({
+      isLoggedIn,
+      favoritePois,
+      isReady: true,
+    });
+  }
+
+  openMasq = () => {
+    Telemetry.add(Telemetry.MASQ_BANNER_CLICK);
+    (new MasqOnboardingModal()).open();
+  }
+
+  closeMasqFooter = () => {
+    Telemetry.add(Telemetry.MASQ_BANNER_CLOSE);
+    localStorage.setItem(`qmaps_v${version}_favorite_masq_footer`, false);
+    this.setState({ masqFooterClosed: true });
+  }
+
+  removeFav = async poi => {
+    Telemetry.add(Telemetry.FAVORITE_DELETE);
+    this.setState(prevState => ({
+      favoritePois: prevState.favoritePois.filter(favorite => favorite !== poi),
+    }));
+    await this.store.del(poi);
+  };
+
+  close = () => {
+    Telemetry.add(Telemetry.FAVORITE_CLOSE);
+    window.app.navigateTo('/');
+  };
+
+  render() {
+    if (!this.state.isReady) {
+      return null;
+    }
+
+    const { favoritePois, isLoggedIn, masqFooterClosed } = this.state;
+    const displayMasqFooter = isMasqEnabled && !isLoggedIn && masqFooterClosed;
+
+    const header = <React.Fragment>
```

Comment:
React.Fragment c'est un truc natof pour faire des mini bouts de JSX embeddables dans un JSX normal?

Reply:
Something like that :) It's a way to group nodes without having to wrap them in a `<div>`, it's useful when you need a single JSX object but you don't want to render meaningless DOM elements.